src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_osa_commands.py

```diff
@@ -153,3 +153,39 @@ def test_openshift_create_default_service_no_aad(self, resource_group, resource_
 
         # delete
         self.cmd('openshift delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+    # It works in --live mode but fails in replay mode.get rid off @live_only attribute once this resolved
+    @live_only()
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitestosa', location='eastus')
+    @ManagedApplicationPreparer()
+    def test_openshift_create_with_monitoring(self, resource_group, resource_group_location, aad_client_app_id, aad_client_app_secret):
+        # kwargs for string formatting
+        osa_name = self.create_random_name('clitestosa', 15)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': osa_name,
+            'location': resource_group_location,
+            'aad_client_app_id': aad_client_app_id,
+            'aad_client_app_secret': aad_client_app_secret,
+            'workspace_id': "test-workspace"
```

Comment:
Is it a workspace name instead of workspace id?

Reply:
It is a workspace resource id.  The workspace name and id are 2 other fields that exist and won't work here.  We are using the example inputs to try to make it clear what kind of id it is.

Reply:
This is the workspace resource id as Jack said.  We are using the same mechanism that we use to force users to grab this id as we do for VNET peering listed in the example inputs.  Trying to mock this means I need a subscription.  I can test that the monitor profile is indeed `None` when creating a cluster.  The cluster will provision as before but with a blank monitoring profile. 

We also do not provide a default log analytics workspace if none is provided like AKS does with its add-on capabilities.

Reply:
Looks like you delete the logic and it's just a renaming PR. Right?

Reply:
I deleted the logic because as of right now we do not want to support workspace name for the reasons @JackQuincy explained.  I am just using this PR to rename the parameters around the flag for documentation purposes. The subsequent work in the future should address these issues.  The user of this flag will be demonstrated within the examples and forced to used the existing workspace in the format that is similar to VNET peering.